src/System.Private.CoreLib/shared/System/Threading/SemaphoreSlim.cs

```diff
@@ -55,7 +43,7 @@ public class SemaphoreSlim : IDisposable
         // The number of synchronously waiting threads, it is set to zero in the constructor and increments before blocking the
         // threading and decrements it back after that. It is used as flag for the release call to know if there are
         // waiting threads in the monitor or not.
-        private int m_waitCount;
+        private volatile int m_waitCount;
```

Comment:
Why does this need to be volatile?

Reply:
It will be accessed by the multiple threads as it keeps the record  of the waiting threads in the monitor

Reply:
It is always accessed under a lock as far as I can tell. Did you happen to trace down when was this volatile added or removed?